### PR TITLE
Binder aware traversal function for context passes

### PR DESCRIPTION
**Pull request by mariari**
Ts PR gives us a generic traversal function that is aware of binders and properly handles each form that might bind.

This means that any pass that cares about closures just needs to supply a proper predicate and transform function.

We also test and document this function in this PR.

### CODE REVIEW

**Review by cyberglot — APPROVED** [review]
LGTM.

**Review by cyberglot — APPROVED** [review]

File: library/Translate/test/Context/Environment.hs [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+capture :: ByteString -> (NameSymbol.T -> Bool) -> IO [Env.Closure']
+capture str trigger = do
+  Right (ctx, _) <-
+    contextualizeFoo str
+  let (_, Cap _ capture) =
+        runCtx (Env.passContextSingle ctx trigger recordClosure) emptyClosure
+  pure capture
+
+-- Right (ctx,_) <- contextualizeFoo "let f = 3"
```
Comment (cwgoes) [inline library/Translate/test/Context/Environment.hs:226]: What are these comments for?

**Review by cwgoes — APPROVED** [review]
ACK modulo nit

File: library/Translate/test/Context/Environment.hs (hunk 1 above)
Comment (mariari) [reply library/Translate/test/Context/Environment.hs:226]: Just for quickly copy and pasting something into the repl

**Pull request merged**